BraveRewardsUI/QA Settings/QASettingsViewController.swift

```diff
@@ -4,207 +4,139 @@
 
 import UIKit
 import BraveRewards
+import BraveShared
+import Shared
+import Static
+
+private typealias EnvironmentOverride = Preferences.Rewards.EnvironmentOverride
 
 /// A special QA settings menu that allows them to adjust debug rewards features
-public class QASettingsViewController: UIViewController {
+public class QASettingsViewController: TableViewController, UITextFieldDelegate {
   
   public let rewards: BraveRewards
   
   public init(rewards: BraveRewards) {
     self.rewards = rewards
-    super.init(nibName: nil, bundle: nil)
+    super.init(style: .grouped)
   }
   
   @available(*, unavailable)
   required init(coder: NSCoder) {
     fatalError()
   }
   
+  private let segmentedControl = UISegmentedControl(items: EnvironmentOverride.allCases.map { $0.name })
+  
+  @objc private func environmentChanged() {
+    let value = segmentedControl.selectedSegmentIndex
+    guard let _ = EnvironmentOverride(rawValue: value) else { return }
+    Preferences.Rewards.environmentOverride.value = value
+    self.rewards.reset()
+    self.showResetRewardsAlert()
+  }
+  
+  private let reconcileTimeTextField = UITextField().then {
+    $0.borderStyle = .roundedRect
+    $0.autocorrectionType = .no
+    $0.autocapitalizationType = .none
+    $0.spellCheckingType = .no
+    $0.returnKeyType = .done
+    $0.textAlignment = .right
+    $0.text = "\(BraveLedger.reconcileTime)"
+    $0.placeholder = "0"
+  }
+  
+  @objc private func reconcileTimeEditingEnded() {
+    guard let value = Int32(reconcileTimeTextField.text ?? "") else {
+      let alert = UIAlertController(title: "Invalid value", message: "Time has been reset to 0 (no override)", preferredStyle: .alert)
+      alert.addAction(.init(title: "OK", style: .default, handler: nil))
+      self.present(alert, animated: true)
+      reconcileTimeTextField.text = "0"
+      BraveLedger.reconcileTime = 0
+      return
+    }
+    BraveLedger.reconcileTime = value
+  }
+  
+  public func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+    textField.resignFirstResponder()
```

Comment:
You might need `endEditing(force: true)` as well?

Reply:
Hmm testing worked fine without that so not sure that I do